MetricsUploader/include/MetricsUploader/MetricsUploader.h

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_METRICS_UPLOADER_METRICS_UPLOADER_H_
+#define ORBIT_METRICS_UPLOADER_METRICS_UPLOADER_H_
+
+#ifdef _WIN32
+#include <windows.h>
+#endif  // WIN32
+
+#include <absl/synchronization/mutex.h>
+
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <string_view>
+
+#include "MetricsUploader/Result.h"
+#include "OrbitBase/Result.h"
+#include "orbit_log_event.pb.h"
+
+namespace orbit_metrics_uploader {
+
+constexpr std::string_view kMetricsUploaderClientDllName = "metrics_uploader_client";
+
+// This class is used for sending log events from Orbit. It works only on Windows and only if
+// metrics_uploader_client.dll is available. The types of logs that could be send are defined in
+// orbit_client_protos::OrbitLogEvent::LogEventType.
+//
+// Only one instance of the class is allowed.
+//
+// Usage example:
+//
+// auto metrics_uploader = MetricsUploader::CreateMetricsUploader();
+// if (metrics_uploader.has_value()) {
+//   metrics_uploader.value()->SendLogEvent(...);
+// }
+//
+class MetricsUploader {
+ public:
+  // Create a MetricsUploader instance when is called first time and ErrorMessage otherwise.
+  [[nodiscard]] static ErrorMessageOr<std::unique_ptr<MetricsUploader>> CreateMetricsUploader(
+      const std::string_view client_name = kMetricsUploaderClientDllName);
+
+  // Unload metrics_uploader_client.dll
+  ~MetricsUploader();
+
+  MetricsUploader(const MetricsUploader& other) = delete;
+  MetricsUploader(MetricsUploader&& other) = default;
+  MetricsUploader& operator=(const MetricsUploader& other) = delete;
+  MetricsUploader& operator=(MetricsUploader&& other) = default;
+
+  // Send log events to the server using metrics_uploader.
+  // Returns true on success and false otherwise.
+  bool SendLogEvent(OrbitLogEvent_LogEventType log_event_type);
+
+ private:
+  // Load metrics uploader client library if available and starts metrics uploader client.
+  MetricsUploader(const std::string_view client_name = kMetricsUploaderClientDllName);
+
+  void Reset();
+  std::string GetErrorMessage(Result error);
+
+  inline static bool is_init_ = false;
+  static absl::Mutex init_mutex_;
```

Comment:
Personal preference: If `is_init_` was inline, I would also make `init_mutex_` inline. Then you don't need the definition in the cpp file.

Reply:
I've changed this code: now I have a non-static mutex which id used for send_log_event, and check if the library was loaded earlier with GetModuleHandleExA function.